src/noisepy/seis/S1_fft_cc_MPI.py

```diff
@@ -300,31 +306,39 @@ def compute_fft(fft_params: ConfigParameters, ch_data: ChannelData) -> NoiseFFT:
 
 
 def _read_channels(
-    executor: Executor, ts: DateTimeRange, store: RawDataStore, channels: List[Channel], samp_freq: int
+    executor: Executor,
+    ts: DateTimeRange,
+    store: RawDataStore,
+    channels: List[Channel],
+    samp_freq: int,
+    single_freq: bool = True,
 ) -> List[Tuple[Channel, ChannelData]]:
     ch_data_refs = [executor.submit(store.read_data, ts, ch) for ch in channels]
     ch_data = _get_results(ch_data_refs)
     tuples = list(zip(channels, ch_data))
-    return _filter_channel_data(tuples, samp_freq)
+    return _filter_channel_data(tuples, samp_freq, single_freq)
```

Comment:
Thanks Carlos,

If I understand this this function right, it returns the channels at the frequency closest to `samp_freq` when `single_freq = True`. While `single_freq = False`, it returns all channels with frequency greater than `samp_freq`. By default, the `fft_param` set this switch as True.

Is there any test needs to be added accordingly?

Reply:
ha I see the tests below now. Looks great!